### PR TITLE
fix(proposal): should show the proposal you submitted and not the previous

### LINKED ISSUE #936: After submitting a proposal draft, you see another proposal
**Affected Portal:**
Select one: DevHub ( probably all )

**Description:**
The two last time I have submitted my contributor reports, when clicking the "Submit draft" button and approving the transaction, I've been taken to another proposal. I have to go to the proposal feed to find my newly submitted proposal draft.

**User Type:**
Specify: Regular User

**Reproduction Steps**
1. Step 1
Create  a proposal like the one here: https://near.social/devhub.near/widget/app?page=proposal&id=207
2. Step 2
Click "Submit draft"
3. Approve the transaction ( "Don't ask again" was not enabled )
4. You are now taken to a different proposal

**Expected behavior**
You should have seen the proposal you just submitted, or be taken to the feed.

**Visual Reference**
If applicable, add screenshots or video recording to help explain your problem.

**Priority**
Select One: Low (P3) – Minor inconvenience

**Environment Details**
- Device: Desktop
- OS: Mac
- Browser Chrome
- Version 129.0.6668.58

### PR DESCRIPTION
**Pull request by petersalomonsen**
After submitting a proposal draft, all the proposal ids are fetched from the contract to check if a new was added. The problem is that NearSocialVM will cache the results of `get_all_proposals_ids` from the previous time that you submitted, and then when cache is invalidated, the first result will return a different number of proposal_ids in case someone submitted a proposal after the last time you did ( which is most often the case ).

To fix this issue, an additional check is added to check that the latest proposal title and description matches what you submitted. This way not only the number of proposal ids is enough to accept that the latest proposal id is the one you submitted, but also it must match the title and description.

This Pull Request adds a playwright test for the scenario, which will break if the fix is reverted.

Here's a video of the test failing, without the fix:

https://github.com/user-attachments/assets/1aa63ea1-78c4-4a38-960d-61e7ad821361

And here's a video of the test passing, with the fix:

https://github.com/user-attachments/assets/a76de7b5-93d5-4b98-b059-66b53c82e0d8

fixes #936

### CODE REVIEW

**Comment by petersalomonsen** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@petersalomonsen Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Weekly streak is on the road, smart strategy! Secure your streak with another PR!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&pr=NEAR-DevHub/neardevhub-bos/942&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&pr=NEAR-DevHub/neardevhub-bos/942&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/petersalomonsen)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/petersalomonsen)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: instances/devhub.near/widget/devhub/entity/proposal/Editor.jsx
Comment (Megha-Dev-19) [inline instances/devhub.near/widget/devhub/entity/proposal/Editor.jsx:455]: can you please make this change for `events` and `infra` proposal and RFP, since there also we'll have the same issue.

**Review by Megha-Dev-19 — APPROVED** [review]
LGTM, left a comment

**Comment by Megha-Dev-19** [comment]
@race-of-sloths score 5

File: instances/devhub.near/widget/devhub/entity/proposal/Editor.jsx
Comment (petersalomonsen) [reply instances/devhub.near/widget/devhub/entity/proposal/Editor.jsx:455]: Thanks for pointing out. I have made a separate issue for `events` and `infra` here: https://github.com/NEAR-DevHub/neardevhub-bos/issues/944

Also pointing out that we need to make sure that tests for creating and submitting a proposal have to be common for all 3 instances.

**Pull request merged**
